apps/rush-lib/src/cli/utilities/pnpm/PnpmLinkManager.ts

```diff
@@ -0,0 +1,190 @@
+
+        const newLocalPackage: BasePackage = BasePackage.createLinkedPackage(
+          dependencyName,
+          matchedVersion,
+          newLocalFolderPath
+        );
+
+        newLocalPackage.symlinkTargetFolderPath = matchedRushPackage.projectFolder;
+        localPackage.children.push(newLocalPackage);
+      } else {
+        // weird state or program bug
+        throw Error('Cannot find rush dependency in rush configuration');
+      }
+    });
+
+    // Iterate through all the regular dependencies
+
+    // With NPM, it's possible for two different projects to have dependencies on
+    // the same version of the same library, but end up with different implementations
+    // of that library, if the library is installed twice and with different secondary
+    // dependencies.The NpmLinkManager recursively links dependency folders to try to
+    // honor this. Since PNPM always uses the same physical folder to represent a given
+    // version of a library, we only need to link directly to that folder, and it will
+    // have a consistent set of secondary dependencies.
+
+    // each of these dependencies should be linked in a special folder that PNPM
+    // creates for the installed version of each .TGZ package, all we need to do
+    // is re-use that symlink in order to get linked to whatever PNPM thought was
+    // appropriate. This folder is usually something like:
+    // C:\{path-to-tgz-with-slashes-replaced-with-"%2F"}\node_modules\{package-name}
+
+    const slashEscapeRegExp: RegExp = new RegExp(path.sep.replace('\\', '\\\\'), 'g');
+    const colonEscapeRegExp: RegExp = new RegExp(':', 'g');
+
+    // e.g.: C%3A%2Fwbt%2Fcommon%2Ftemp%2Fprojects%2Fapi-documenter.tgz
+    const escapedPathToTgzFile: string = path.join(
+      this._rushConfiguration.commonTempFolder,
+      'projects',
+      `${unscopedTempProjectName}.tgz`)
+        .replace(slashEscapeRegExp, '%2F')
+        .replace(colonEscapeRegExp, '%3A');
+
+    // tslint:disable-next-line:max-line-length
+    // e.g.: C:\wbt\common\temp\node_modules\.local\C%3A%2Fwbt%2Fcommon%2Ftemp%2Fprojects%2Fapi-documenter.tgz\node_modules
+    const pathToLocalInstallation: string = path.join(
+      this._rushConfiguration.commonTempFolder,
+      RushConstants.nodeModulesFolderName,
+      '.local',
+      escapedPathToTgzFile,
+      RushConstants.nodeModulesFolderName);
+
+    Object.keys(commonPackage.packageJson!.dependencies || {}).forEach((dependencyName: string) => {
+      // the dependency we are looking for should have already created a symlink here
+
+      const dependencyLocalInstallationSymlink: string = path.join(
+        pathToLocalInstallation,
+        dependencyName);
+
+      if (!fsx.existsSync(dependencyLocalInstallationSymlink)) {
+        throw Error(`Cannot find installed dependency "${dependencyName}" in "${pathToLocalInstallation}"`);
```

Comment:
>throw [](start = 8, length = 5)

Can you add comments something like:

// If this happens, it's a bug in the Rush algorithm or an unexpected behavior of the PNPM tool